Source/WTF/wtf/URLHelpers.cpp

```diff
@@ -651,7 +651,7 @@ std::optional<String> mapHostName(const String& hostName, URLDecodeFunction deco
     auto expectedSourceBuffer = string.charactersWithNullTermination();
     if (!expectedSourceBuffer)
         return std::nullopt;
-    auto sourceBuffer = expectedSourceBuffer.value();
+    Vector<UChar> sourceBuffer(WTFMove(*expectedSourceBuffer));
```

Comment:
Maybe we could use span?
`auto sourceBuffer =  expectedSourceBuffer.value().span();`